client/src/App.js

```diff
@@ -6,6 +6,91 @@ import './App.css';
 const url = process.env.NODE_ENV === 'development' ? "ws://localhost:5001" : "ws://codos.nessenbu.com:5001";
 const connection = new WebSocket(url);
 
+class UserNameModal extends Component {
+  constructor(props) {
+    super(props);
+    this.textInput = React.createRef();
+    this.handleClick = this.handleClick.bind(this);
+    this.callback = props.callback
+  }
+
+  handleClick = function() {
+    this.callback(this.textInput.current.value);
+  }
+
+  render() {
+    return (
+      <div className="user-name-prompt">
+        <p> Input ur name: </p>
+        <input type="text" ref={this.textInput}/><br/>
+        <input
+          type="button"
+          value="Submit"
+          onClick={ this.handleClick }
+        />
+      </div>
+    )
+  }
+}
+
+class Players extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      redPlayers: props.redPlayers === undefined ? [] : props.redPlayers,
+      bluePlayers: props.bluePlayers === undefined ? [] : props.bluePlayers,
+      playerId: props.playerId
+    };
+
+    this.swapTeams = this.swapTeams.bind(this);
+  }
+
+  // See the TODO below in the Tile component :3
+  componentWillReceiveProps(props) {
+    this.setState({
+      redPlayers: props.redPlayers === undefined ? [] : props.redPlayers,
+      bluePlayers: props.bluePlayers === undefined ? [] : props.bluePlayers
+    });
+  }
+
+  swapTeams = () => {
+    connection.send(JSON.stringify({ action: "swapTeams", playerId: this.state.playerId}));
+  }
+
+  render() {
+    var redPlayers = [];
+    var bluePlayers = [];
+
+    for (const player of this.state.redPlayers) {
+      redPlayers.push(<p className="score-red">{player.name}</p>);
+    }
```

Comment:
this could be written functionally with a `map`

```
const redPlayers = this.state.redPlayers.map(player => <p className="score-red">{player.name}</p>);
```

Reply:
oh yeah and also, you can use `const` and still modify the array, you just can't change the reference

Reply:
tbh, I would have done that to start if it was in java. I'm not used to js syntax for these sorts of things